lib/libc/syslog/lib_lowsyslog.c

```diff
@@ -150,9 +150,9 @@ int lowvsyslog(int priority, FAR const char *fmt, va_list ap)
 #if defined(CONFIG_LOGM) && defined(CONFIG_SYSLOG2LOGM)
 		ret = logm_internal(LOGM_LOWPUT, LOGM_UNKNOWN, priority, fmt, ap);
 #else
-		irqstate_t flags = enter_critical_section();
```

Comment:
May I know the reason to change from enter/leave_critical_section to irqsave/irqrestore here? 
As for now, we are still not very sure which scenario to apply which. In RTK BSP level, there is also quite some places will have to enable/disable interrupts for certain operation. As far as I understood, enter/leave_critical_section will have consideration for multicore coherency (using spinlock), could you share any specific scenario we should apply those?

Reply:
enter_critical_section API is only available in kernel side code. 
So, if the library is built for app side, then it generates a build error. 
However, this particular API of lowsyslog will be used only in kernel side. So, I have modified the code to use enter_critical_section if we are building the library for kernel.


In a general case, we need to replace all irqsave / restore by enter / leave_critical_section for SMP. I will check this in case of TizenRT code. Please check rtk code and replace the irqsave api in all locations.

Reply:
Understood, I will check and raise it in another PR.